### PR TITLE
VCST-1555: Minor improvements and bugfixes

### PR DESCRIPTION
**Pull request by tatarincev**
### Jira-link:
https://virtocommerce.atlassian.net/browse/VCST-1555

### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.Skyflow_3.807.0-pr-10-c860.zip

### CODE REVIEW

File: src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs [hunk 1]
```diff
@@ -13,4 +14,10 @@ public SkyflowCardQueryBuilder(IMediator mediator, IAuthorizationService authori
         : base(mediator, authorizationService)
     {
     }
+
+    protected override SkyflowCardQuery GetRequest(IResolveFieldContext<object> context)
+    {
+        var result = base.GetRequest(context);
+        return result;
+    }
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs:22]: This override doesn't change the base behavior

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using Microsoft.AspNetCore.Authorization;
 using VirtoCommerce.ExperienceApiModule.Core.BaseQueries;
 using VirtoCommerce.Skyflow.XApi.Schemas;
+using GraphQL;
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs:5]: Undo

File: src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs [hunk 3]
```diff
@@ -12,5 +13,5 @@ public class SkyflowCardQueryBuilder : QueryBuilder<SkyflowCardQuery, SkyflowCar
     public SkyflowCardQueryBuilder(IMediator mediator, IAuthorizationService authorizationService)
         : base(mediator, authorizationService)
     {
-    }
+    }
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Skyflow.XApi/Queries/SkyflowCardQueryBuilder.cs:16]: Undo

File: src/VirtoCommerce.Skyflow.Data/Services/SkyflowClient.cs [hunk 4]
```diff
@@ -151,7 +153,18 @@ private async Task<SkyflowBearerTokenResponse> GetBearerTokenInternal(SkyflowSer
...
     }
-
+    private string GetQueryString(object callParams)
+    {
+        var result = string.Empty;
+        if (callParams != null)
+        {
+            var type = callParams.GetType();
+            var props = type.GetProperties();
+            var pairs = props.Select(x => x.Name + "=" + x.GetValue(callParams, null)).ToArray();
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Skyflow.Data/Services/SkyflowClient.cs:163]: Name and value should be encoded.
Use [Microsoft.AspNetCore.Http.Extensions.QueryBuilder](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.extensions.querybuilder?view=aspnetcore-8.0).

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

**Review by artem-dudarev — APPROVED** [review]

**Comment by ekuvirto** [comment]
https://github.com/VirtoCommerce/vc-module-skyflow/releases/tag/3.805.1

**Pull request merged**